src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -656,42 +684,43 @@ public boolean shouldSkipMonitoredTripCheck(boolean persist) throws Exception {
             // checking today's date at the earliest in case the user has paused trip monitoring for a while
             targetZonedDateTime = trip.tripZonedDateTime(DateTimeUtils.nowAsLocalDate());
 
-            // Check if the journeyState indicates that an itinerary has already been calculated in a previous run of
-            // this CheckMonitoredTrip. If the targetDate is null, then the current date has not yet been checked. If
-            // the journeyState's targetDate is not null, that indicates that today has already been checked.
-            // Therefore, for recurring trips, advance targetDate by another day before calculating when the next
-            // itinerary occurs. One-time trips will have their status set to PAST_TRIP and will be skipped.
-            if (previousJourneyState.targetDate != null && !trip.isOneTime()) {
-                LocalDate lastDate = DateTimeUtils.getDateFromString(
-                    previousJourneyState.targetDate,
-                    DateTimeUtils.DEFAULT_DATE_FORMAT_PATTERN
-                );
+            // Attempt to advance to the next monitored day, except for one-time trips or if tracking is ongoing.
```

Comment:
This method is already quite large, is it possible to put this addition in a separate method?

Reply:
Yes, extracted in https://github.com/ibi-group/otp-middleware/pull/228/commits/a1b3eacb806d0625c18efd374c2a6577ac637a78.